wiring/src/spark_wiring_string.cpp

```diff
@@ -217,7 +217,8 @@ String & String::copy(const char *cstr, unsigned int length)
 		return *this;
 	}
 	len = length;
-	strcpy(buffer, cstr);
+	memcpy(buffer, cstr, length);
+	buffer[len] = 0;
```

Comment:
The problem isn't null terminating the local buffer, but that it's copying too much data into the buffer.  We can't null terminate the `cstr` at the given length since that's input data (and const).